src/main/java/com/musala/atmosphere/client/Device.java

```diff
@@ -1337,14 +1341,40 @@ public void startScreenRecording() {
      *        - the maximum recording duration in minutes
      */
     public void startScreenRecording(int timeLimit) {
+        screenRecordUploadDiectoryName = getScreenRecordUploadDiectoryName();
         communicator.sendAction(RoutingAction.START_RECORDING, timeLimit);
     }
 
     /**
      * Stops screen recording.
      */
     public void stopScreenRecording() {
-        communicator.sendAction(RoutingAction.STOP_RECORDING);
+        communicator.sendAction(RoutingAction.STOP_RECORDING, screenRecordUploadDiectoryName);
+    }
+
+    private String getScreenRecordUploadDiectoryName() {
+        String uploadDirectoryName = "";
```

Comment:
You don't need to declare this parameter here. When you try to get the property you will receive empty String if there is no value in the properties file.